aiohttp/web_routedef.py

```diff
@@ -36,7 +39,7 @@
 
 class AbstractRouteDef(abc.ABC):
     @abc.abstractmethod
-    def register(self, router: UrlDispatcher) -> None:
+    def register(self, router: UrlDispatcher) -> Optional[AbstractRoute]:
```

Comment:
This is basically the part I have the most questions about. So far, I've made this return value `Optional` (and everywhere else where its return value is propagated). The reason for this is because of the StaticDef and its `register` implementation. Since `add_static` returns an `AbstractResource`, do you think it would be sensible for this to be a union of `AbstractRoute` and `AbstractResource` for that reason?

Reply:
`StaticResource` internally has `self._routes` which is a dict of `ResourceRoute` instances.
Perhaps we can extend `resource.get_info()` to return routes mapping; it can avoid access to private members.